### PR TITLE
content: News entry for the CERN Open Data Repository release

### PR DESCRIPTION
**Pull request by psaiz**

### CODE REVIEW

File: cernopendata/modules/fixtures/data/docs/cern-open-data-release-2024/cern-open-data-release-2024.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+* Change the self-maintained ElasticSearch instance by OpenSearch instances provided by the CERN OpenSearch service
+* Redesign facets for results filtering
+* Changes in the search functionality, including modifying the default search to do an `AND` of the terms (instead of `OR`), escaping the special character `/` (allowing searches by DOI)
+
+
+
+## Search tips
+The search engine of the CERN Open Data Portal allows different types of queries. The most common filters can be applied with the selection boxes on the left: experiment, type of record, file format, etc...
+
+The search bar offers more flexibility, and it allows to search by any terms. By default, it will do and `AND` of all
```
Comment (zzacharo) [inline cernopendata/modules/fixtures/data/docs/cern-open-data-release-2024/cern-open-data-release-2024.md:18]:
```suggestion
The search bar offers more flexibility, and it allows to search by any terms. By default, it will do an `AND` of all
```

File: cernopendata/modules/fixtures/data/docs/cern-open-data-release-2024/cern-open-data-release-2024.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+the terms that have been introduced. For instance, a search for `heavy ion` will return only documents that contain both
+words. The default behaviour can be modified using the reserved word `AND`, `OR` and the parenthesis `(`, `)`. Note that
```
Comment (zzacharo) [inline cernopendata/modules/fixtures/data/docs/cern-open-data-release-2024/cern-open-data-release-2024.md:20]: maybe we add an example query here?

File: cernopendata/modules/fixtures/data/docs/cern-open-data-release-2024/cern-open-data-release-2024.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+On top of that, it is also possible to search on specific fields. For instance, take a look at the following queries:
+
+* `authors.orcid:0000-0002-4485-2972`: returns all the documents from that author.
+* `10.7483/OPENDATA.CMS.W26R.J96R`: returns all documents with those terms in any field.
+* `doi:10.7483/OPENDATA.CMS.W26R.J96R`: returns the document with that Digital Object Identifier (doi)
+* `doi:10.7483*`: returns all the documents with a doi starting with those values.
+
+
+The syntax for these queries is defined by the OpenSearch query_string, which is based in the [Apache Lucene syntax](https://lucene.apache.org/core/2_9_4/queryparsersyntax.html)
```
Comment (zzacharo) [inline cernopendata/modules/fixtures/data/docs/cern-open-data-release-2024/cern-open-data-release-2024.md:43]: I see that you have linked the syntax but maybe we could have some practical examples here for easy-to-use templates for our users. WDYT?

File: cernopendata/modules/fixtures/data/docs/cern-open-data-release-2024/cern-open-data-release-2024.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    "collections": [
+      {
+        "primary": "News"
+      }
+    ],
+    "date_published": "2024-01-31",
+    "short_description": {
+      "content": "CERN Open Data release 2024"
+    },
+    "slug": "cern-open-data-release-2024",
```
Comment (tiborsimko) [inline cernopendata/modules/fixtures/data/docs/cern-open-data-release-2024/cern-open-data-release-2024.json:17]: The SLUG says "cern-open-data-release-2024" and the title says "CERN Open Data Release 2024". This could be easily misunderstand as presenting an open data release rather than presenting a repository infrastructure upgrade. Moreover, there may be more repository updates coming later in the year 2024, so we should leave the room for those, and be more descriptive about this change. For example: "Upgrade of the CERN Open Data portal web interface" or "Upgrade of search functionality".

Basically, let's pick some leading news time (which could be Angular->React or the improved search), and let's model the title and the slug after that. In this way, the news item will be more end-user-facing. And the other "minor" news times would come along as a by-product without particular highlighting.

File: cernopendata/modules/fixtures/data/docs/cern-open-data-release-2024/cern-open-data-release-2024.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
+A new version of the CERN Open Data Repository has been deployed in January 2024. The changes were deployed first in the [CERN Open Data QA instance](https://opendata-qa.cern.ch), and then propagated to the production instance.
```
Comment (tiborsimko) [inline cernopendata/modules/fixtures/data/docs/cern-open-data-release-2024/cern-open-data-release-2024.md:1]: We should not advertise the QA instance address to the world from the PROD instance. Recall the preparatory JADE work, CMS 2016 work happening on QA. Moreover, users of the open data portal don't really care much whether this was deployed somewhere beforehand, they care more about which changes this upgrade brings, so I would focus on that and remove this phrase.

File: cernopendata/modules/fixtures/data/docs/cern-open-data-release-2024/cern-open-data-release-2024.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+The list of changes include:
+
+* Change of technology, from Angular to ReactJS and Bootstrap[3,4] to Semantic UI for the web interface
```
Comment (tiborsimko) [inline cernopendata/modules/fixtures/data/docs/cern-open-data-release-2024/cern-open-data-release-2024.md:5]: - User interface was changed from Angular to React which improved rendering on mobile devices.  

Basically, it's good to always end the user-oriented release news with a little "so what?" message explaining to the target audience why this may be interesting for them.

File: cernopendata/modules/fixtures/data/docs/cern-open-data-release-2024/cern-open-data-release-2024.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+* Update the dependencies on the invenio framework modules
```
Comment (tiborsimko) [inline cernopendata/modules/fixtures/data/docs/cern-open-data-release-2024/cern-open-data-release-2024.md:6]: This news item is also formulated in a very internal manner, the eople would not know what this change brings, and would wonder "what this brings in for me?". I would leave it out (similarly to the docker  news) or else explain what the Invenio module upgrade brings.  (E.g. "improving exposure of records by sitemaps" to come up with a simple example.)

File: cernopendata/modules/fixtures/data/docs/cern-open-data-release-2024/cern-open-data-release-2024.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+* Move to xrootd version 5
+* Move to docker compose version 2
+* Change the self-maintained ElasticSearch instance by OpenSearch instances provided by the CERN OpenSearch service
```
Comment (tiborsimko) [inline cernopendata/modules/fixtures/data/docs/cern-open-data-release-2024/cern-open-data-release-2024.md:9]: I would phrase the ES to OS change again to be more user-centric. Data scientists do not really care about underlying search technology and where it is hosted, they care more that this change is improving some search nuisances that they might have had with the old system. So I would rather highlight something like:

- Improvements to the search functionality making the returned query results to be more relevant.
- Improvements to the faceting functionality allowing to combine many new facet categories together.

I guess you get what I mean?

File: cernopendata/modules/fixtures/data/docs/cern-open-data-release-2024/cern-open-data-release-2024.md (hunk 3 above)
Comment (tiborsimko) [reply cernopendata/modules/fixtures/data/docs/cern-open-data-release-2024/cern-open-data-release-2024.md:43]: The seach tips are nice, but might totally dominate the news item. (Which may be a good thing, see my remark above about choosing a highlight reel news and inventing title and slug around them.) 

Ideally I would say we should create static a "Search tips" page that would be linked from the search box so that users can easily discover the syntax later on without reading the news announcement.  And in this manner the news announcement can stay brief and simply link to that same help page.

File: cernopendata/modules/fixtures/data/docs/cern-open-data-release-2024/cern-open-data-release-2024.json [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+    "body": {
+      "content": "cern-open-data-release-2024.md",
+      "format": "md"
+    },
+    "collections": [
+      {
+        "primary": "News"
+      }
+    ],
+    "date_published": "2024-01-31",
```
Comment (tiborsimko) [inline cernopendata/modules/fixtures/data/docs/cern-open-data-release-2024/cern-open-data-release-2024.json:13]: Please add also `"featured": 1,` so that the news item will appear on the home page, and decreased `"featured"` values on other existing records.

See example: https://github.com/cernopendata/opendata.cern.ch/pull/3503/files

File: cernopendata/modules/fixtures/data/docs/cern-open-data-release-2024/cern-open-data-release-2024.md (hunk 3 above)
Comment (psaiz) [reply cernopendata/modules/fixtures/data/docs/cern-open-data-release-2024/cern-open-data-release-2024.md]: I've split the document in two

File: cernopendata/modules/fixtures/data/docs/cern-open-data-release-2024/cern-open-data-release-2024.md (hunk 6 above)
Comment (psaiz) [reply cernopendata/modules/fixtures/data/docs/cern-open-data-release-2024/cern-open-data-release-2024.md]: ok, done

File: cernopendata/modules/fixtures/data/docs/cern-open-data-release-2024/cern-open-data-release-2024.md (hunk 7 above)
Comment (psaiz) [reply cernopendata/modules/fixtures/data/docs/cern-open-data-release-2024/cern-open-data-release-2024.md]: Using the same logic, the xrootd part should also be out. That would be a bit too much. What about combining the three of them into a single entry of update dependencies?  People do know that dependencies tend to be get obsolete, and htat it is good to be to the recent versions

File: cernopendata/modules/fixtures/data/docs/cern-open-data-release-2024/cern-open-data-release-2024.md (hunk 8 above)
Comment (psaiz) [reply cernopendata/modules/fixtures/data/docs/cern-open-data-release-2024/cern-open-data-release-2024.md]: Ok, rephrased pointing out that the version that we were running was not supported anymore.

File: cernopendata/modules/fixtures/data/docs/cod-portal-upgrade-31-1-2024/cod-portal-upgrade-31-1-2024.json [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+      {
+        "primary": "News"
+      }
+    ],
+    "date_published": "2024-01-31",
+    "featured": 1,
+    "short_description": {
+      "content": "New version of the CERN Open Data Portal deployed"
+    },
+    "slug": "cod-portal-upgrade-31-1-2024",
```
Comment (tiborsimko) [inline cernopendata/modules/fixtures/data/docs/cod-portal-upgrade-31-1-2024/cod-portal-upgrade-31-1-2024.json:18]: We are using YYYY-MM-DD everywhere, so if you want to use this slug, better say `cod-portal-upgrade-20240131`.

File: cernopendata/modules/fixtures/data/docs/cod-portal-upgrade-31-1-2024/cod-portal-upgrade-31-1-2024.json [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+    "title": "Upgrade of the CERN Open Data portal",
```
Comment (tiborsimko) [inline cernopendata/modules/fixtures/data/docs/cod-portal-upgrade-31-1-2024/cod-portal-upgrade-31-1-2024.json:19]: I would still prefer something more targeted such as "Improvements to the search functionality" as explained before. It's more descriptive than the generic "upgrade" only.

File: cernopendata/modules/fixtures/data/docs/cod-portal-upgrade-31-1-2024/cod-portal-upgrade-31-1-2024.md [hunk 12]
```diff
@@ -0,0 +1,14 @@
+A new version of the CERN Open Data Repository has been deployed in January 2024.
+Some of the highlights of this release are:
+
+* Change of technologies, from Angular to [ReactJS](https://react.dev/) and from Bootstrap to [Semantic UI](https://semantic-ui.com/)
```
Comment (tiborsimko) [inline cernopendata/modules/fixtures/data/docs/cod-portal-upgrade-31-1-2024/cod-portal-upgrade-31-1-2024.md:4]: Official name is React, not ReactJS.  And I don't think it is necessary to mention Semantic UI, the open data users don't really care about whether we use Semantic UI or Material UI.  We could even remove Angular/React and just say "Improvements to the web interface...".

File: cernopendata/modules/fixtures/data/docs/cod-portal-upgrade-31-1-2024/cod-portal-upgrade-31-1-2024.md [hunk 13]
```diff
@@ -0,0 +1,14 @@
...
+for the web interface, which improves the navigation and overall use experience.
+* Changes in the search functionality, including modifying the default search to do an `AND` of the terms (instead of `OR`),
```
Comment (tiborsimko) [inline cernopendata/modules/fixtures/data/docs/cod-portal-upgrade-31-1-2024/cod-portal-upgrade-31-1-2024.md:6]: What about Improvements instead of Changes to highlight the _positive_ aspects of changes?

File: cernopendata/modules/fixtures/data/docs/cod-portal-upgrade-31-1-2024/cod-portal-upgrade-31-1-2024.md [hunk 14]
```diff
@@ -0,0 +1,14 @@
...
+escaping the special character `/` (allowing searches by DOI).
+* Redesign facets for results filtering.
+* Change the self-maintained ElasticSearch instance, which had reached its end of life support, by the open-sourced
```
Comment (tiborsimko) [inline cernopendata/modules/fixtures/data/docs/cod-portal-upgrade-31-1-2024/cod-portal-upgrade-31-1-2024.md:9]: I don't think it is necessary to mention. We could also say that we changed from Python 2.7 to Python 3.6 due to end-olf-life etc.  These kind of news are internal not relevant to the open data users because they don't see them.

File: cernopendata/modules/fixtures/data/docs/cod-portal-upgrade-31-1-2024/cod-portal-upgrade-31-1-2024.md [hunk 15]
```diff
@@ -0,0 +1,14 @@
...
+OpenSearch instances provided by the CERN OpenSearch service.
+* Update to the latest version of the underlying components, including the invenio framework, xrootd and docker compose.
```
Comment (tiborsimko) [inline cernopendata/modules/fixtures/data/docs/cod-portal-upgrade-31-1-2024/cod-portal-upgrade-31-1-2024.md:11]: Again, open data users don't see "docker compose". The open data curators do, but these know to read the DEVELOPING guide, that we changed already. 

It is good to distinguish between (i) end-user facing news and (ii) developer-facing news. I think this item should describe the former, and we have developer guide to describe to latter.

File: cernopendata/modules/fixtures/data/docs/cod-portal-upgrade-31-1-2024/cod-portal-upgrade-31-1-2024.json (hunk 10 above)
Comment (psaiz) [reply cernopendata/modules/fixtures/data/docs/cod-portal-upgrade-31-1-2024/cod-portal-upgrade-31-1-2024.json:18]: Done, thanks

File: cernopendata/modules/fixtures/data/docs/cod-portal-upgrade-31-1-2024/cod-portal-upgrade-31-1-2024.json (hunk 11 above)
Comment (psaiz) [reply cernopendata/modules/fixtures/data/docs/cod-portal-upgrade-31-1-2024/cod-portal-upgrade-31-1-2024.json:19]: As suggested in the previous comment as well, I've split the initial proposal into two documents: one for the search, one for the upgrade. I would prefer to keep the message of the generic one with the upgrade, since it does more changes than just the search funcionality

File: cernopendata/modules/fixtures/data/docs/cod-portal-upgrade-31-1-2024/cod-portal-upgrade-31-1-2024.md (hunk 12 above)
Comment (psaiz) [reply cernopendata/modules/fixtures/data/docs/cod-portal-upgrade-31-1-2024/cod-portal-upgrade-31-1-2024.md]: For the react vs reactjs, it seems to be known by both names. 
Regarding removing also react/semantic, we could indeed remove all the highlights and just leave one-liner saying that we have improved the web interface.  In my opinion, that would be too simplistic, and giving more information might be appreciated by the user.

File: cernopendata/modules/fixtures/data/docs/cod-portal-upgrade-31-1-2024/cod-portal-upgrade-31-1-2024.md (hunk 13 above)
Comment (psaiz) [reply cernopendata/modules/fixtures/data/docs/cod-portal-upgrade-31-1-2024/cod-portal-upgrade-31-1-2024.md]: Done

File: cernopendata/modules/fixtures/data/docs/cod-portal-upgrade-31-1-2024/cod-portal-upgrade-31-1-2024.md (hunk 15 above)
Comment (psaiz) [reply cernopendata/modules/fixtures/data/docs/cod-portal-upgrade-31-1-2024/cod-portal-upgrade-31-1-2024.md]: Since the developers will also read this, it might be worth mentioning some of the changes here. Otherwise, we could just forget about the highlights and put a single line saying `improving web interface`. 
Then, I find it difficult where to set the line. Some of the things are indeed quite internal (like python version). Others, like the fact that opendata uses the invenio framework, I find it that it is very good to remind our users. The docker seems to be the one were we have more conflicting points of view. What about we leave docker out, and all the other in?

**Review by zzacharo — APPROVED** [review]
